sympy/sets/conditionset.py

```
@@ -196,57 +200,39 @@ def free_symbols(self):
-                    if base != self.base_set:
-                        # it will be aggravating to have the dummy
-                        # symbol change if you are trying to target
-                        # the base set so if the base set is changed
-                        # leave the dummy symbol alone -- a second
-                        # subs will be needed to change the dummy
-                        return self.func(sym, cond, base)
-                    else:
-                        return self.func(new, cond.subs(old, new), base)
-                raise ValueError(filldedent('''
-                    A dummy symbol can only be
-                    replaced with a symbol having the same
-                    assumptions or one having a single assumption
-                    having the same commutativity.
-                '''))
-            # don't target cond: it is there to tell how
-            # the base set should be filtered and if new is not in
-            # the base set then this substitution is ignored
-            return self.func(sym, cond, base)
-        else:
-            cond = self.condition.subs(old, new)
-            base = self.base_set.subs(old, new)
-            # The condition may have become true due to assumptions
-            # on 'sym'. In order for .subs() to be consistent with
-            # __new__ we *don't* check if 'sym' actually belongs to
-            # 'base'. In other words: assumptions are ignored.
-            return self.func(self.sym, cond, base)
+        insym = old in flatten([sym])
+        # prioritize changing a symbol in the base
+        newbase = base.subs(old, new)
+        if newbase != base:
+            if not insym:
+                # XXX should we ignore the subs if TypeError is
+                # raised and just return self? e.g. if cond is
+                # x < 1 then subs(x, I) raises TypeError
+                cond = cond.subs(old, new)
```

I think it's better just to let the exception raise unless we can think of a good reason for wanting to do such a substitution.